packages/nimble-components/src/select/styles.ts

```diff
@@ -28,6 +40,81 @@ export const styles = css`
     [part='end'] {
         display: contents;
     }
+
+    .listbox {
+        overflow-x: clip;
+    }
+
+    .listbox.empty slot {
+        display: none;
+    }
+
+    .listbox.above {
+        flex-flow: column-reverse;
+    }
+
+    .filter-field {
+        display: flex;
+        flex-direction: row;
+        align-items: center;
+        height: ${controlHeight};
+        background: transparent;
+    }
+
+    .filter-icon {
+        padding-left: ${smallPadding};
+    }
+
+    .filter-field::after,
+    .filter-field::before {
+        content: '';
+        position: absolute;
+        height: 0px;
+        border-bottom: rgba(${borderRgbPartialColor}, 0.1) 2px solid;
+    }
+
+    .filter-field::after {
+        top: calc(${controlHeight} + ${smallPadding} - ${borderWidth});
+    }
+
+    .filter-field:before {
+        bottom: calc(${controlHeight} + ${smallPadding} - ${borderWidth});
+    }
+
+    .filter-field:not(.above)::after,
+    .filter-field.above::before {
+        width: calc(100% - (2 * ${borderWidth}));
+    }
```

Comment:
Can you group these based on [dom order](https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md) and target element instead of similar styles?


![image](https://github.com/ni/nimble/assets/1588923/119db136-8f9a-46e0-a304-50513fa70bcf)


```
.filter-field {}
.filter-field::before {}
.filter-field.above::before {}
.filter-field::after {}
.filter-field:.above::after {}

.filter-icon {}
.filter-input {}
.filter-input::placeholder {}
.filter-input:focus-visible{}
```